File: c/two-fer/two_fer.c
#include "two_fer.h"

void two_fer(char *buffer, const char *name) {
  const char format[] = "One for %s, one for me.";
  const char unknown_name_pronoun[] = "you";
  sprintf(buffer, format, name != NULL ? name : unknown_name_pronoun);
}

File: go/two-fer/two_fer.go
package twofer

import "fmt"

// ShareWith generates a sharing message in the format
// "One for <name>, one for me.". If the name is empty,
// it defaults to "you".
func ShareWith(name string) string {
	const defaultPronoun string = "you"
	const shareMessageFormat string = "One for %s, one for me."

	if name != "" {
		return fmt.Sprintf(shareMessageFormat, name)
	}

	return fmt.Sprintf(shareMessageFormat, defaultPronoun)
}
